pkg/reconciler/managed/reconciler.go

```diff
@@ -941,9 +955,10 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (resu
 
 	// If we started but never completed creation of an external resource we
 	// may have lost critical information. For example if we didn't persist
-	// an updated external name we've leaked a resource. The safest thing to
-	// do is to refuse to proceed.
-	if meta.ExternalCreateIncomplete(managed) {
+	// an updated external name which is non-deterministic, we have leaked a
+	// resource. The safest thing to do is to refuse to proceed. However, if
+	// the resource has a deterministic external name, it is safe to proceed.
+	if meta.ExternalCreateIncomplete(managed) && !r.deterministicExternalName {
```

Comment:
WDYT about debug logging this case (but not returning an error) even when there's a deterministic external name?

Reply:
Sure, something like this https://github.com/crossplane/crossplane-runtime/pull/841/commits/6120cd37b63623eae817c75fc6efe893288621f6?

Reply:
Yes that looks great.